tests/bluetooth/gatt/src/main.c

```diff
@@ -142,6 +142,185 @@ ZTEST(test_gatt, test_gatt_unregister)
 		     "Test service unregister failed");
 }
 
+/* Test that a service A can be re-registered after registering it once, unregistering it, and then
+ * registering another service B.
+ * No pre-allocated handles. Repeat the process multiple times.
+ */
+ZTEST(test_gatt, test_gatt_reregister)
+{
+	struct bt_gatt_attr local_test_attrs[] = {
+		/* Vendor Primary Service Declaration */
+		BT_GATT_PRIMARY_SERVICE(&test_uuid),
+
+		BT_GATT_CHARACTERISTIC(&test_chrc_uuid.uuid, BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE,
+				       BT_GATT_PERM_READ_AUTHEN | BT_GATT_PERM_WRITE_AUTHEN,
+				       read_test, write_test, test_value),
+	};
+
+	struct bt_gatt_attr local_test1_attrs[] = {
+		/* Vendor Primary Service Declaration */
+		BT_GATT_PRIMARY_SERVICE(&test1_uuid),
+
+		BT_GATT_CHARACTERISTIC(&test1_nfy_uuid.uuid, BT_GATT_CHRC_NOTIFY, BT_GATT_PERM_NONE,
+				       NULL, NULL, &nfy_enabled),
+		BT_GATT_CCC(test1_ccc_cfg_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+	};
+	struct bt_gatt_service local_test_svc = BT_GATT_SERVICE(local_test_attrs);
+	struct bt_gatt_service local_test1_svc = BT_GATT_SERVICE(local_test1_attrs);
+
+	/* Checks that the procedure is successful for a few iterations. However, a single iteration
+	 * was sufficient to trigger the reported bug.
+	 */
+	for (int i = 0; i < 10; i++) {
+
+		zassert_false(bt_gatt_service_register(&local_test_svc),
+			      "Test service A registration failed");
+
+		zassert_false(bt_gatt_service_unregister(&local_test_svc),
+			      "Test service A unregister failed");
+
+		/* Check that the handles are the same as before registering the service */
+		for (int j = 0; j < local_test_svc.attr_count; j++) {
+			zassert_equal(local_test_svc.attrs[j].handle, 0x0000,
+				      "Test service A handle not reset");
+		}
```

Comment:
Shouldn't this loop exist before the first call to `bt_gatt_service_register(&local_test_svc)` as well? You refer to "before registering the service", but we do no verify that these are 0x0000 before registering the service